src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -298,7 +299,8 @@ private void visitParameterDef(DetailAST parameterDef) {
 
         if (grandParentAST.getType() == TokenTypes.METHOD_DEF
             && isCheckedMethod(grandParentAST)
-            && isVerifiable(grandParentAST)) {
+            && isVerifiable(grandParentAST)
+            && !AnnotationUtil.containsAnnotation(grandParentAST, "Override")) {
```

Comment:
As discussed in main issue, this should be in the check for the entire method.
Maybe this should be moved to the `isCheckedMethod` method?